app/src/main/java/com/hedvig/app/feature/offer/OfferViewModel.kt

```diff
@@ -199,4 +215,21 @@ class OfferViewModelImpl(
             response.getOrNull()?.data?.let { offerRepository.removeStartDateFromCache(quoteIds, it) }
         }
     }
+
```

Comment:
I think you can notify the view here with some kind of event instead of returning from the function. Can make it tricky to follow if the error result notifies the view but the success result returns as usual.

Reply:
I agree - I'll remake this code to use a more event-based architecture.

I noticed that doing it that way will fix an issue related to the use of `lifecycleScope` too (which dies when the activity dies, ie when rotating the screen/resizing the window).

Additionally, I noticed that we have an issue with how the events are architected today. I'll make a proposal on how to change it and present it to you when it's done 🙂